modules/snooker/src/main/ts/ephox/snooker/model/RunOperation.ts

```diff
@@ -74,27 +74,8 @@ export interface ExtractPasteRows {
 const fromWarehouse = (warehouse: Warehouse, generators: Generators) =>
   Transitions.toGrid(warehouse, generators, false);
 
-const deriveRows = (rendered: Structs.RowDetails[], generators: Generators) => {
```

Comment:
This was the core of the issue, as it's the part that would generate a new row whenever all the cells in a row were modified. The rest is largely just flow on effects from having to pass the row details around. It did end up making this logic much simpler :shrug: